src/layout/DwindleLayout.cpp

```diff
@@ -144,13 +151,14 @@ void CHyprDwindleLayout::applyNodeDataToWindow(SDwindleNodeData* pNode, bool for
         return;
     }
 
-    PWINDOW->m_sSpecialRenderData.rounding = true;
-    PWINDOW->m_sSpecialRenderData.border   = true;
-    PWINDOW->m_sSpecialRenderData.decorate = true;
+    PWINDOW->m_sSpecialRenderData.rounding   = WORKSPACE_RULE.rounding.value_or(true);
+    PWINDOW->m_sSpecialRenderData.decorate   = WORKSPACE_RULE.decorate.value_or(true);
+    PWINDOW->m_sSpecialRenderData.border     = WORKSPACE_RULE.border.value_or(true);
+    PWINDOW->m_sSpecialRenderData.borderSize = borderSize;
```

Comment:
`value_or`? Why are you hard-setting it here? It should be left as `-1` (unset) if it's not overridden.

Reply:
Isn't that what the previous code was doing? I hard coded it just to replicate the previous behaviour, which was:
```cpp
PWINDOW->m_sSpecialRenderData.rounding = true;
PWINDOW->m_sSpecialRenderData.border   = true;
PWINDOW->m_sSpecialRenderData.decorate = true;
```

Reply:
yes, borderSize. it should be -1 if a workspace rule is not overriding it. Currently, it's always set.